packages/pools/src/concentrated/fetch-tick-data-provider.ts

```diff
@@ -0,0 +1,185 @@
+import { Dec, Int } from "@keplr-wallet/unit";
+import {
+  estimateInitialTickBound,
+  LiquidityDepth,
+  maxTick,
+  minTick,
+} from "@osmosis-labs/math";
+
+import {
+  ConcentratedLiquidityPool,
+  TickDataProvider,
+  TickDepths,
+} from "./pool";
+
+type TickDepthsResponse = {
+  liquidity_depths: {
+    liquidity_net: string;
+    tick_index: string;
+  }[];
+};
+
+/** Default tick data provider that fetches ticks for a single CL pool with `fetch` if the environment supports it, if not a fetcher can be supplied. */
+export class FetchTickDataProvider implements TickDataProvider {
+  protected _zeroForOneTicks: LiquidityDepth[] = [];
+  protected _oneForZeroTicks: LiquidityDepth[] = [];
+
+  protected _zeroForOneBoundIndex = new Int(0);
+  protected _oneForZeroBoundIndex = new Int(0);
+
+  constructor(
+    protected readonly baseNodeUrl: string,
+    protected readonly poolId: string,
+    protected readonly nextTicksRampMultiplier = new Int(9),
```

Comment:
consider moving these default values to a constant

Reply:
See constants rant above